docs/source/user-guide/vision/VisionData.rst

```diff
@@ -0,0 +1,184 @@
+
+.. code-block:: python
+
+    from deepchecks.vision import VisionData, BatchOutputFormat
+
+    def deepchecks_map(batch) -> BatchOutputFormat:
+        # Extracting images and label and converting images of (N, C, H, W) into (N, H, W, C)
+        images = batch[0].permute(0, 2, 3, 1)
+        labels = batch[1]
+        # Convert ImageNet format images into to [0, 255] range format images.
+        mean, std  = [0.485, 0.456, 0.406], [0.229, 0.224, 0.225]
+        images = np.clip(std * images.numpy() + mean, 0, 1) * 255
+        return BatchOutputFormat(images= images, labels= labels)
+
+    deepchecks_dataset = my_dataset.map(deepchecks_map)
+    vision_data = VisionData(deepchecks_dataset, task_type='classification')
+    vision_data.head()
+
+
+
+Adding Model Predictions
+========================
+Some of deepchecks tests, including the :doc:`model evaluation checks and suite </checks_gallery/vision>`,
+require model predictions in
+order to run. Models prediction are supplied via the batch loader in a similar fashion to the images and labels.
+There are several ways to supply them which can be divide into two categories: Pre-calculated predictions and
+on-demand inference.
+
+On-demand Inference
+-------------------
+In this case we will need to incorporate the model object in the relevant format transformation function. This can be
+done either by using the model as a global variable, creating a wrapper class for the transformation function or
+creating a closure function. We will demonstrate the last option via the pytorch interface.
+
+.. code-block:: python
+
+    import torch
+    from torch.utils.data import DataLoader
+    from deepchecks.vision import VisionData, BatchOutputFormat
+
+    def create_deepchecks_collate(model):
+        def deepchecks_collate(data) -> BatchOutputFormat:
+            # Extracting images and label and predicting using the model
+            raw_images = torch.stack([x[0] for x in data])
+            predictions = model(images)
+            labels = [x[1] for x in data]
+            # Convert ImageNet format images into to [0, 255] range format images.
+            mean, std  = [0.485, 0.456, 0.406], [0.229, 0.224, 0.225]
+            images = np.clip(std * raw_images.permute(0, 2, 3, 1).numpy() + mean, 0, 1) * 255
+            return BatchOutputFormat(images= images, labels= labels, predictions= predictions)
+        return deepchecks_collate
+
+    data_loader = DataLoader(my_dataset, batch_size=64, generator=torch.Generator(),
+                             collate_fn=create_deepchecks_collate(my_model))
+    vision_data = VisionData(data_loader, task_type='classification')
+    vision_data.head()
+
+.. _vision_data__precalculated_predictions:
+Pre-calculated Predictions
+--------------------------
+It is specifically recommended to use this option if your model object is unavailable locally (for example if
```

Comment:
```suggestion
It is recommended to use this option if your model object is unavailable locally (for example if
```
Would actually rephrase further - 
"This is the recommended option, as pre-calculating the predictions saves time during the checks' run, and is more efficient as sometimes you would need to rerun your checks to see if you were able to fix your data issues."

Or something like that. And put this option first.

Why? Because if the user does this outside of the check, they would think our checks are faster